analysis/rogue/src/InstantPoison.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import SPELLS from 'common/SPELLS';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import BoringValueText from 'parser/ui/BoringValueText';
+import { SpellIcon, SpellLink } from 'interface';
+import { formatPercentage } from 'common/format';
+import { NumberThreshold, ThresholdStyle, When } from 'parser/core/ParseResults';
+
+class InstantPoison extends Analyzer {
+  numPoisonHits: number = 0;
+  numMeleeHits: number = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.INSTANT_POISON), this.onInstantPoisonDamage);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.MELEE), this.onMeleeDamage);
+  }
+
+  onInstantPoisonDamage(event: DamageEvent) {
+    this.numPoisonHits += 1;
+  }
+
+  onMeleeDamage(event: DamageEvent) {
+    this.numMeleeHits += 1;
+  }
+
+  get procPercentage(): number {
+    return this.numPoisonHits / this.numMeleeHits;
+  }
+
+  get instantPoisonSuggestionThresholds(): NumberThreshold {
+    return {
+      actual: this.procPercentage,
+      isLessThan: {
+        minor: 0.27,
+        average: 0.25,
+        major: 0.2,
+      },
+      style: ThresholdStyle.PERCENTAGE,
+    }
+  }
+
+  statistic(): React.ReactNode {
+    return (
+      <Statistic
+        size="flexible"
+        category={STATISTIC_CATEGORY.GENERAL}
+      >
+        <BoringValueText label={<><SpellIcon id={SPELLS.INSTANT_POISON.id}/> Instant Poison Proc Percentage</>}>
+          {formatPercentage(this.procPercentage)} %
+        </BoringValueText>
+      </Statistic>
+    );
+  }
+
+  suggestions(when: When) {
+    when(this.instantPoisonSuggestionThresholds)
+      .addSuggestion((suggest, actual, recommended) => suggest(
+        <>
+          Ensure that your poisons are freshly applied before the fight starts so that they do not expire.
+        </>)
+        .icon(SPELLS.INSTANT_POISON.icon)
+        .actual(<>You procced <SpellLink id={SPELLS.INSTANT_POISON.id} /> on {formatPercentage(this.procPercentage)} % of your melee hits.</>)
+        .recommended(`Unless unlucky, you should have around a 30% proc chance over the fight.`)
```

Comment:
If luck can have a significant impact, consider reducing your threshold instead. In general though proc chances average out very nicely, so I think it might be better to omit this text and put the actual recommended value here.

Reply:
Sorry for the extremely long delay in responding. I'll swap out the text and lower the threshold in the next revision.